rpc/src/error.rs

```diff
@@ -1,290 +1,223 @@
+            }
+            | e | {
+                format_args!("parse error: {}", e.reason)
+            },
+
+        Server
+            {
+                reason: String
+            }
+            | e | {
+                format_args!("server error: {}", e.reason)
+            },
+
+        ClientInternal
+            {
+                reason: String
+            }
+            | e | {
+                format_args!("client internal error: {}", e.reason)
+            },
+
+        Timeout
+            {
+                duration: Duration
+            }
+            | e | {
+                format_args!(
+                    "timed out waiting for healthy response after {}ms",
+                    e.duration.as_millis()
+                )
+            },
+
+        ChannelSend
+            | _ | { "failed to send message to internal channel" },
+
+        InvalidUrl
+            { url: Url }
+            | e | {
+                format_args!(
+                    "cannot use URL {} with HTTP clients",
+                    e.url
+                )
+            },
+
+        InvalidUri
+            [ InvalidUriError ]
+            | _ | { "invalid URI" },
+
+        Tendermint
+            [ tendermint::Error ]
+            | _ | { "tendermint error" },
+
+        ParseInt
+            [ DisplayOnly<std::num::ParseIntError> ]
+            | _ | { "error parsing integer" },
+
+        OutOfRange
+            [ DisplayOnly<std::num::TryFromIntError> ]
+            | _ | { "number out of range" },
+
+        InvalidNetworkAddress
+            | _ | { "only TCP-based node addresses are supported" },
+
+        MismatchResponse
+            | _ | { "no matching response for incoming request" },
+
+        UnrecognizedEventType
+            {
+                event_type: String
+            }
+            | e | {
+                format_args!("unrecognized event type: {}", e.event_type)
+            },
+
+        Serde
+            [ DisplayOnly<serde_json::Error> ]
+            | _ | { "serde parse error" },
+
+        ParseUrl
+            [ DisplayOnly<url::ParseError> ]
+            | _ | { "parse error" },
+
+        Tungstenite
+            [ TungsteniteError ]
+            | _ | { "tungstenite error" },
+
+        Join
+            [ JoinError ]
+            | _ | { "join error" },
+
+        MalformedJson
+            | _ | { "server returned malformatted JSON (no 'result' or 'error')" },
+
+        UnsupportedScheme
+            {
+                scheme: String
+            }
+            | e | {
+                format_args!("unsupported scheme: {}", e.scheme)
+            },
+
+        UnsupportedRpcVersion
+            {
+                version: String,
+                supported: String
+            }
+            | e | {
+                format_args!("server RPC version unsupported: '{}' (only '{}' supported)",
+                    e.version, e.supported)
+            },
+
+    }
+}
+
+impl Clone for Error {
```

Comment:
We can't use `#[derive(Clone)]` here because `eyre::Report` doesn't implement `Clone`. This is a general problem for deriving on the main error type, because error tracers like `eyre::Report` do not implement many common traits, not even `std::error::Error`. So I made `flex-error` leave that to the user to decide how to deal with the tracer part.